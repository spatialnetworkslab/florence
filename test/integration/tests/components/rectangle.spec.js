/// <reference types="Cypress" />

context('Assertions', () => {
  beforeEach(() => {
    cy.visit('/rectangles')
  })

  describe('Rectangle Tests', () => {
    it('[render] rectangles rendered', () => {
      cy.get('.rectangle').should('have.length', 3)
    })

    // it('[render] rectangles snapshot correctly', () => {
    //   cy.get('svg')
    //     .toMatchSnapshot()
    // })

    it('[interaction] clicking on rectangle changes color', () => {
      var rectangle = cy.get('.rectangle').first()
      rectangle.should('have.attr', 'fill', 'yellow')
      rectangle.click()
      rectangle.should('have.attr', 'fill', 'blue')
    })
  })
})
