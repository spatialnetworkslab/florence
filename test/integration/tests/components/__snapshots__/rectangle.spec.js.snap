exports[`Assertions > Rectangle Tests > [render] rectangles snapshot correctly #0`] = `
<svg
  width="500"
  height="500"
  viewBox="0 0 500 500"
  preserveAspectRatio="xMidYMid meet"
>
  <defs>
    <clipPath id="clip-sc0">
      <rect x="50" y="50" width="400" height="400"></rect>
    </clipPath>
    <clipPath id="clip-sc0-data">
      <rect x="50" y="50" width="400" height="400" fill="white"></rect>
    </clipPath>
  </defs>
  <g class="section" clip-path="url(#clip-sc0)">
    <g class="rectangle-layer">
      <path
        class="rectangle"
        d="M75,50L175,50L175,450L75,450Z"
        fill="yellow"
        stroke="none"
        stroke-width="0"
        opacity="1"
      ></path>
      <path
        class="rectangle"
        d="M200,50L300,50L300,263.33L200,263.33Z"
        fill="yellow"
        stroke="none"
        stroke-width="0"
        opacity="1"
      ></path>
      <path
        class="rectangle"
        d="M325,50L425,50L425,183.33L325,183.33Z"
        fill="yellow"
        stroke="none"
        stroke-width="0"
        opacity="1"
      ></path>
    </g>
  </g>
</svg>
`;
